app/dashboard/src/layouts/DriveBar.tsx

```diff
@@ -84,14 +96,43 @@ export default function DriveBar(props: DriveBarProps) {
           }
         : {}),
       newProject: () => {
-        doCreateProject()
+        setIsCreatingProject(true)
+        doCreateProject(
+          null,
+          null,
+          project => {
+            setCreatedProjectId(project.projectId)
+          },
+          () => {
+            setIsCreatingProject(false)
+          }
+        )
       },
       uploadFiles: () => {
         uploadFilesRef.current?.click()
       },
     })
   }, [isCloud, doCreateDirectory, doCreateProject, inputBindings])
 
+  const createdProjectQuery = useQuery<Project>(
+    createdProjectId
+      ? createGetProjectDetailsQuery.createPassiveListener(createdProjectId)
+      : { queryKey: ['__IGNORE__'], queryFn: skipToken }
+  )
+
+  const isFetching =
+    (createdProjectQuery.isLoading ||
+      (createdProjectQuery.data &&
+        createdProjectQuery.data.state.type !== ProjectState.opened &&
+        createdProjectQuery.data.state.type !== ProjectState.closing)) ??
+    false
+
+  React.useEffect(() => {
```

Comment:
Oh, we are supposed to remove these kinds of useEffect, aren't we?

Reply:
I guess we can use isFetching directly, right?

Reply:
maybe, but we do need to clear the loading state somehow otherwise the button will show as loading when the user opens the project again. so if we don't set isCreatingProject to false here then we'd still need to set createdProjectId to `null` here

Reply:
Can we use `isFetching` instead of `isCreatingProject` ?

Reply:
i don't think so. as mentioned above:
- if we don't unset `createdProjectId` then the listener continues listening. when the user reopens the same project, the button will be disabled out of nowhere because the state is no longer `opened` nor `closing`.
- also note that before we receive `createdProjectId` from callbacks, `isFetching` is false because we have no query, but the button *must* be disabled to prevent spam clicking